File: scripts/get-streamlit-version.py
#!/usr/bin/env python3
import re
import sys
from pathlib import Path

def get_streamlit_version():
    """Extract the Streamlit version from streamlit/lib/setup.py"""
    setup_py_path = Path(__file__).parent.parent / "streamlit" / "lib" / "setup.py"
    
    with open(setup_py_path, 'r') as f:
        content = f.read()
    
    match = re.search(r'VERSION = "([^"]+)"', content)
    if not match:
        raise ValueError("Could not find VERSION in streamlit/lib/setup.py")
    
    return match.group(1)

if __name__ == "__main__":
    print(get_streamlit_version())

File: scripts/get-streamlit-version.js
#!/usr/bin/env node
import { execSync } from 'node:child_process';
import path from 'node:path';
import { fileURLToPath } from 'node:url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));

export function getStreamlitVersion() {
  try {
    const version = execSync('python3 scripts/get-streamlit-version.py', {
      cwd: path.resolve(__dirname, '..'),
      encoding: 'utf8'
    }).trim();
    return version;
  } catch (error) {
    throw new Error(`Failed to get Streamlit version: ${error.message}`);
  }
}

if (import.meta.url === `file://${process.argv[1]}`) {
  console.log(getStreamlitVersion());
}
